src/tools/genpolicy/rules.rego

```diff
@@ -176,11 +182,28 @@ state_allows(key, value) = action {
   action := null
 }
 
+# delete key=value from state
+state_del_key(key) = action {
+  print("state_del_key: ", key)
+  state := get_state()
+  print("state_del_key: deleting from state key =", key)
+  path := get_state_path(key)
+  action := {
+    "op": "remove",
+    "path": path,
+  }
+}
+
 # helper functions to interact with the state
 get_state() = state {
   state := data["pstate"]
 }
 
+get_state_val(key) = value {
+    state := get_state()
+    value := state[key]
+}
```

Comment:
Did you consider inlining this to

```rego
get_state()[key]
```

Reply:
I will look into this. Thanks @burgerdev

Reply:
I have not inlined this for now.